src/vtkh/filters/ParticleAdvection.cpp

```diff
@@ -0,0 +1,90 @@
+#include <iostream>
+#include <vtkh/filters/ParticleAdvection.hpp>
+#include <vtkm/filter/ParticleAdvection.h>
+#include <vtkh/vtkh.hpp>
+#include <vtkh/Error.hpp>
+
+#if VTKH_PARALLEL
+#include <vtkm/thirdparty/diy/diy.h>
+#include <vtkm/thirdparty/diy/mpi-cast.h>
+#include <mpi.h>
+#endif
+
+namespace vtkh
+{
+
+ParticleAdvection::ParticleAdvection()
+{
+}
+
+ParticleAdvection::~ParticleAdvection()
+{
+
+}
+
+void ParticleAdvection::PreExecute()
+{
+  Filter::PreExecute();
+  Filter::CheckForRequiredField(m_field_name);
+}
+
+void ParticleAdvection::PostExecute()
+{
+  Filter::PostExecute();
+}
+
+void ParticleAdvection::DoExecute()
+{
+  this->m_output = new DataSet();
+
+#ifndef VTKH_BYPASS_VTKM_BIH
+
+#ifdef VTKH_PARALLEL
+  // Setup VTK-h and VTK-m comm.
+  MPI_Comm mpi_comm = MPI_Comm_f2c(vtkh::GetMPICommHandle());
+  vtkm::cont::EnvironmentTracker::SetCommunicator(vtkmdiy::mpi::communicator(vtkmdiy::mpi::make_DIY_MPI_Comm(mpi_comm)));
+#endif
+
+  const int num_domains = this->m_input->GetNumberOfDomains();
+
+  vtkm::cont::PartitionedDataSet inputs;
+  for(int i = 0; i < num_domains; ++i)
+  {
+    vtkm::Id domain_id;
+    vtkm::cont::DataSet dom;
+    this->m_input->GetDomain(i, dom, domain_id);
+    if(dom.HasField(m_field_name))
+    {
+      using vectorField_d = vtkm::cont::ArrayHandle<vtkm::Vec<vtkm::Float64, 3>>;
+      using vectorField_f = vtkm::cont::ArrayHandle<vtkm::Vec<vtkm::Float32, 3>>;
+      auto field = dom.GetField(m_field_name).GetData();
+      if(!field.IsType<vectorField_d>() && !field.IsType<vectorField_f>())
+      {
+        throw Error("Vector field type does not match <vtkm::Vec<vtkm::Float32,3>> or <vtkm::Vec<vtkm::Float64,3>>");
+      }
+    }
+    else
+    {
+      throw Error("Domain does not contain specified vector field for ParticleAdvection analysis.");
```

Comment:
We keep running into weird data sets. My only concern is that one data set of many might not have the field, and this would fail. An alternative might be to create a new data set that only contains data sets that have the field in them, and the error would only be if none of the data sets have the field.

Reply:
Hmm. I do know that VTKm will check these as well.  The vtkm filter will fail if not everything has the field.. As such, would it make sense to have VTKm validate this? And so just remove this check?